apache2/waf_logging/waf_log_util.cc

```diff
@@ -167,12 +220,34 @@ void free_json(char* str) {
     std::free(str);
 }
 
-void init_appgw_rules_id_hash() {
-    std::ifstream ifs("/etc/nginx/modsec/disableableRules.conf");
+static rule_ids_vector read_disableable_rule_ids(const std::string& filename) {
+    rule_ids_vector rule_ids;
+    std::ifstream ifs(filename);
     if(ifs) {
         std::copy(std::istream_iterator<int>(ifs),
                 std::istream_iterator<int>(),
-                std::back_inserter(default_disableable_rule_ids));
-        std::sort(default_disableable_rule_ids.begin(), default_disableable_rule_ids.end());
+                std::back_inserter(rule_ids));
+        std::sort(std::begin(rule_ids), std::end(rule_ids));
+    }
+
+    return rule_ids;
+}
+
+void init_appgw_rules_id_hash() {
+    std::map<rule_set_info, std::string> pathsMap {
+        {OWASP_2_2_9, "/etc/nginx/modsec/OWASP/2.2.9/disableableRules.conf"},
+        {OWASP_3_0, "/etc/nginx/modsec/OWASP/3.0/disableableRules.conf"},
+        {OWASP_3_1, "/etc/nginx/modsec/OWASP/3.1/disableableRules.conf"},
+    };
+    for (const auto& item : pathsMap) {
+        auto vec = read_disableable_rule_ids(item.second);
+        if (!vec.empty()) {
+            disableable_rule_ids_map[item.first] = vec;
+        }
+    }
+    if (disableable_rule_ids_map.size() > 0) {
+        // If we have found type/version-specific files, we don't need to read the default (deprecated one)
+        return;
     }
+    default_disableable_rule_ids = read_disableable_rule_ids("/etc/nginx/modsec/disableableRules.conf");
```

Comment:
What are the default ruleIds? Which version of CRS do these refer to?

Reply:
This is the old file for the case (currently in use) when we only ever have a single WAF policy for the entire gateway. In this case this file contains rule IDs for whatever CRS version is being used by this WAF policy.

When we introduce support for multiple WAF policies, this file will no longer be used. The code here remains for backward compatibility.

I did not come up with a name better than "default" for this one but if you have suggestions I am willing to rename it to something that better reflects its gist and its purpose. "old_.."? "deprecated_..."?

Reply:
Rename to current_rule_ids?